src/Messages/Interactions/TSOutgoingMessage.m

```diff
@@ -8,44 +8,93 @@
 #import "TSAttachmentStream.h"
 #import "TSContactThread.h"
 #import "TSGroupThread.h"
+#import <YapDatabase/YapDatabase.h>
+#import <YapDatabase/YapDatabaseTransaction.h>
 
 NS_ASSUME_NONNULL_BEGIN
 
+NSString *const kTSOutgoingMessageSentRecipientAll = @"kTSOutgoingMessageSentRecipientAll";
+
+@interface TSOutgoingMessage ()
+
+@property (atomic) TSOutgoingMessageState messageState;
+@property (atomic) BOOL hasSyncedTranscript;
+@property (atomic) NSString *customMessage;
+@property (atomic) NSString *mostRecentFailureText;
+@property (atomic) BOOL wasDelivered;
+// For outgoing, non-legacy group messages sent from this client, this
+// contains the list of recipients to whom the message has been sent.
+//
+// This collection can also be tested to avoid repeat delivery to the
+// same recipient.
+@property (atomic) NSArray<NSString *> *sentRecipients;
+
+@property (atomic) TSGroupMetaMessage groupMetaMessage;
+
+@end
+
+#pragma mark -
+
 @implementation TSOutgoingMessage
 
+@synthesize sentRecipients = _sentRecipients;
+
 - (instancetype)initWithCoder:(NSCoder *)coder
 {
     self = [super initWithCoder:coder];
+
     if (self) {
         if (!_attachmentFilenameMap) {
             _attachmentFilenameMap = [NSMutableDictionary new];
         }
+        
+        // Migrate message state.
+        if (_messageState == TSOutgoingMessageStateSent_OBSOLETE) {
+            _messageState = TSOutgoingMessageStateSentToService;
+        } else if (_messageState == TSOutgoingMessageStateDelivered_OBSOLETE) {
+            _messageState = TSOutgoingMessageStateSentToService;
+            _wasDelivered = YES;
+        }
+        if (!_sentRecipients) {
+            _sentRecipients = [NSArray new];
+        }
     }
+
     return self;
 }
 
 - (instancetype)initWithTimestamp:(uint64_t)timestamp
 {
+    OWSAssert(0);
+
     return [self initWithTimestamp:timestamp inThread:nil];
 }
 
 - (instancetype)initWithTimestamp:(uint64_t)timestamp inThread:(nullable TSThread *)thread
 {
+    OWSAssert(0);
+
     return [self initWithTimestamp:timestamp inThread:thread messageBody:nil];
 }
 
 - (instancetype)initWithTimestamp:(uint64_t)timestamp
                          inThread:(nullable TSThread *)thread
                       messageBody:(nullable NSString *)body
 {
-    return [self initWithTimestamp:timestamp inThread:thread messageBody:body attachmentIds:[NSMutableArray new]];
+    return [self initWithTimestamp:timestamp
+                          inThread:thread
+                       messageBody:body
+                     attachmentIds:[NSMutableArray new]
+                  expiresInSeconds:0];
 }
 
 - (instancetype)initWithTimestamp:(uint64_t)timestamp
                          inThread:(nullable TSThread *)thread
                       messageBody:(nullable NSString *)body
                     attachmentIds:(NSMutableArray<NSString *> *)attachmentIds
 {
+    OWSAssert(0);
+
```

Comment:
Is this like a deprecation warning? Dare you just delete the method?

or at least add a comment.